test/package_config_impl_test.dart

```diff
@@ -140,6 +143,19 @@ void main() {
       expect(resolved, root.resolve("a/b"));
     });
   });
+
+  test("PackageConfig.writeString()", () {
+    var package = Package("name", root);
+    var original = PackageConfig([package], extraData: unique);
+    var buffer = StringBuffer();
+    PackageConfig.writeString(original, buffer);
+    expect(json.decode(buffer.toString()), {
+      "configVersion": 2,
+      "packages": [
+        {"name": "name", "rootUri": "file:///tmp/root/"}
+      ]
```

Comment:
Should this contain `extraData`?

Reply:
It can't. The `unique` value is just `Object`, which isn't JSON-serializable. For the `extraData` to be written, it must be a `Map<String,_>` and the values themselves must be JSON-serializable. Anything which isn't is just ignored when writing to JSON.